python/pyspark/sql/functions.py

```diff
@@ -1847,8 +1848,7 @@ def __call__(self, *cols):
         return Column(jc)
 
 
-@since(1.3)
-def udf(f, returnType=StringType()):
+def _udf(f, returnType=StringType()):
     """Creates a :class:`Column` expression representing a user defined function (UDF).
 
     .. note:: The user-defined functions must be deterministic. Due to optimization,
```

Comment:
Even though we've got this with an _ maybe through in the docstring its for internal use only? Also since this is for internal use only maybe most of this should be moved to the public udf function now instead of on the internal only method?

Unless we intend for users to call `_udf` in which case we need to give it a different name.